desc/objectives/_coils.py

```diff
@@ -1182,8 +1198,23 @@ def __init__(
         self._eq = eq
         self._field = field
         self._field_grid = field_grid
-        self._vacuum = vacuum
-        things = [field]
+        self._vacuum = vacuum or qfm_surface
+        self._qfm_surface = qfm_surface
+        errorif(
+            qfm_surface and hasattr(eq, "L_lmn"),
```

Comment:
would prefer an actual instance check here instead of relying on attributes to be/not be there